internal/index/indexoperations/index_test.go

```diff
@@ -64,3 +54,65 @@ func TestListIndexes(t *testing.T) {
 		t.Errorf("output does not match: %s", diff)
 	}
 }
+
+func TestAddIndex(t *testing.T) {
+	type args struct {
+		index Index
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "invalid git repository",
+			args: args{
+				index: Index{
+					Name: "foo",
```

Comment:
Is that even a valid case? The cobra command already ensures, that both index name and index URL are given. So we should not bother about calling this function with an empty URL, right?

Reply:
You're right this is kind of confusing, I wanted to add a case where the provided url is incorrect. In this case the test doesn't initialize an empty repo since the url is blank. I'll change this.

Reply:
How do you tell whether a URL is correct / incorrect?

Reply:
I'm using the `localRepo` variable to initialize an empty git repo that gets used as the url in clone. In the success case it initializes a git repo at `local/foo` and then clones that to `index/foo`, one of the failure cases initializes the repo at `index/foo` already so that it tests the case where an index with a name already exists.

I'm pretty much using the existence of the local git repo as a correct url.

Reply:
I see, good idea. So then I recommend to separate the two test cases into two test functions. That makes the tests so much more readable!